<?xml version="1.0"?>
<launch>
  <arg name="footprint_file" default="ERROR: did not specify argument --footprint_file--" />
  <arg name="scan_topics" default="ERROR: did not specify argument --scan_topics--" />
  <arg name="cmd_vel_in" default="/cmd_vel_safe" />
  <arg name="cmd_vel_out" default="/cmd_vel" />
  <arg name="target_frame" default="/base_link" />

  <node pkg="mcr_collision_velocity_filter" type="collision_velocity_filter_node" name="collision_velocity_filter" ns="mcr_navigation" output="screen">
	<rosparam file="$(arg footprint_file)" command="load"/>
	<rosparam file="$(find mcr_collision_velocity_filter)/ros/config/collision_velocity_filter.yaml" command="load"/>

	<rosparam param="scan_topics" subst_value="True">$(arg scan_topics)</rosparam>
	<rosparam param="target_frame" subst_value="True">$(arg target_frame)</rosparam>

	<remap from="~cmd_vel_in" to="$(arg cmd_vel_in)" />
	<remap from="~cmd_vel_out" to="$(arg cmd_vel_out)" />    
  </node>

</launch>
